src/Compilers/Shared/GlobalAssemblyCacheHelpers/GacFileResolver.cs

```diff
@@ -17,13 +17,10 @@ namespace Microsoft.CodeAnalysis.Scripting.Hosting
     /// </summary>
     internal sealed class GacFileResolver : IEquatable<GacFileResolver>
     {
-        // Consider a better availability check (perhaps the presence of Assembly.GlobalAssemblyCache once CoreCLR mscorlib is cleaned up).
-        // https://github.com/dotnet/roslyn/issues/5538
-
         /// <summary>
-        /// Returns true if GAC is available on the platform.
+        /// Returns true if GAC is available on the current platform.
         /// </summary>
-        public static bool IsAvailable => CoreClrShim.AssemblyLoadContext.Type == null;
+        public static bool IsAvailable => typeof(object).Assembly.GlobalAssemblyCache;
```

Comment:
was this done because it is a perf improvement?

Reply:
No, it's done because it's a better (direct) check.